packages/lib/storage/src/new/write.ts

```diff
@@ -0,0 +1,126 @@
+import { Envelop } from 'dm3-lib-messaging';
+import {
+    MAX_CONVERATION_ENTRIES_PER_CHUNK,
+    MAX_MESSAGE_SIZE,
+} from './constants';
+import { getConversationListKey, getMessageChunkKey } from './keys';
+import {
+    getAccountManifest,
+    getConversationListChunk,
+    getConversationManifest,
+    getMessageChunk,
+} from './read';
+import {
+    AccountManifest,
+    ConversationList,
+    ConversationManifest,
+    Db,
+    MessageChunk,
+} from './types';
+import { getSize } from './utils';
+
+/**
+ * This function adds a new conversation to the conversation list.
+ * It will not directly write to the storage, but instead return the updated conversation list and account manifest.
+ *
+ * @param {string} contactEnsName - The ENS name of the contact.
+ * @param {Db} db - The database object, which contains the account's ENS name and the sign function.
+ * @returns {Promise<{conversationList: ConversationList, accountManifest: AccountManifest}>}
+ *  A promise that resolves to an object containing the updated conversation list and account manifest.
+ */
+export async function addConversation(
+    contactEnsName: string,
+    db: Db,
+): Promise<{
+    conversationList: ConversationList;
+    accountManifest: AccountManifest;
+}> {
+    // get the account manifest and calculate the target chunk index based on the current conversation list counter.
+    const accountMainfest = await getAccountManifest(db);
+    const targetChunkIndex = Math.floor(
+        accountMainfest.conversationListCounter /
+            MAX_CONVERATION_ENTRIES_PER_CHUNK,
+    );
+
+    // get the conversation list chunk at the target chunk index.
+    const conversationListChunk = await getConversationListChunk(
+        db,
+        targetChunkIndex,
+    );
+
+    // calculate the key for the conversation list chunk and creates a new conversation list
+    // with the new conversation added
+    const key = await getConversationListKey(db, targetChunkIndex);
+
+    return {
+        conversationList: {
+            key,
+            conversationList: conversationListChunk
+                ? [...conversationListChunk.conversationList, contactEnsName]
```

Comment:
that would put the contactEnsName at the end of the conversationList. Could it make sene to insert it at the beginning so latest conversations are fetched at first when paginated?